Nightwatch Integration for Functional Tests

justafish opened the pull request.

Commit:
nightwatch

Commit:
nightwatch updates

Commit:
do not start chromedriver on ci env

justafish committed:
Shift files around and have a separate environment for the testbot

justafish committed:
Merge branch 'nightwatch' of github.com:mattgrill/drupal into nightwatch

mattgrill committed:
Merge pull request #1 from justafish/nightwatch

Shift files around and have a separate environment for the testbot

Commit:
linting cleanup

justafish committed:
Doesn't need a separate ci command as nightwatch is already running in a container

justafish committed:
fix path

mattgrill committed:
Merge pull request #2 from justafish/nightwatch

Doesn't need a separate ci command as nightwatch is already running i…

justafish committed:
fight me

justafish committed:
srsly

mattgrill committed:
Merge pull request #3 from justafish/nightwatch

fight me

dawehner committed:
add a way to configure the hostname

justafish committed:
add babel environments

mattgrill committed:
Merge pull request #4 from dawehner/nightwatch

add a way to configure the hostname

mattgrill committed:
Merge pull request #5 from justafish/nightwatch

add babel environments

Commit:
do not adjust old codebase with commonjs modules settings

Commit:
temporarily remove babel-register

Commit:
adjust args to actually pass correctly.

Commit:
actually use chromeArgs array

Commit:
custom command for adding baseurl prefix

dawehner commented on core/tests/Drupal/Nightwatch/Tests/index.js:
```diff
@@ -1,9 +1,9 @@
   'Demo Drupal.org': (browser) => {
     browser
-      .url('https://www.drupal.org/')
+      .drupalURL('/community')
```
What about using "relativeUrl" or something like that? Using Drupal makes me shrug here.

mattgrill replied:
Good point. I'll change it.

Commit:
use relativeURL over drupalURL

dawehner committed:
Apply 2926633

dawehner committed:
Fix test to actually work against a drupal site

Commit:
use nightwatch.settings.json for config. needs docs

dawehner committed:
Improve initial setup

dawehner committed:
Use the install command

dawehner committed:
Wire everything together

dawehner committed:
Create a setup file

dawehner committed:
fix the test setup code

mattgrill committed:
Merge pull request #3 from jsdrupal/apply-patch

Apply patch from https://www.drupal.org/project/drupal/issues/2926633

larowlan committed:
Issue #2865184 by chr.fritsch, phenaproxima, larowlan, marcoscano, xjm: Allow MediaSource plugins provide default field form/view display settings

larowlan committed:
Issue #2928256 by marcoscano, seanB: Users shouldn't be able to change the media source plugin after the media type is created

larowlan committed:
Issue #2795317 by hswong3i, alexpott, Lendude, bircher, dawehner, martin107, Jo Fitzgerald, mondrake: Allow PHPUnit 6+ support for object mocking

larowlan committed:
Issue #2928249 by alexpott, mondrake: Introduce a PHPUnit 6+ compatibility layer for Drupal\Tests\Listeners classes

Commit:
Issue #2930197 by mondrake, amateescu: EntityDefinitionUpdateTest fails with contrib db driver (again)

mattgrill committed:
Merge pull request #4 from jsdrupal/full-test-with-install

Full test with install combined -> nightwatch

Commit:
Issue #2627512 by gambry, jhedstrom, mpdonadio, Jo Fitzgerald, jibran, tedbow, bkosborne, alexpott, vprocessor, xjm: Datetime Views plugins don't support timezones

Commit:
Issue #2929464 by tedbow, mpdonadio: Tests under "core/modules/ckeditor/tests/modules/src/Kernel" are in the wrong folder and do not get tested

Commit:
Issue #2928778 by plach: Exception when trying to save a new revision after manually setting the original revision ID

Commit:
Issue #2928522 by yo30, dawehner: \Drupal\FunctionalJavascriptTests\WebDriverWebAssert misses some deprecations

Commit:
Issue #2894014 by adriancid, Cottser: Additional space in a field setting schema label

larowlan committed:
Issue #2880445 by pjcdawkins, japerry, gargsuchi, q0rban: Config sync should not throw a warning when not being writable

webchick committed:
Issue #2928702 by Wim Leers, borisson_, tedbow: Make EntityResourceTestBase's field_rest_test_multivalue test field less invasive: omit it from normalizations

webchick committed:
Issue #2927566 by Wim Leers: Unit test EntityReferenceFieldItemNormalizerTest mocks incorrectly

larowlan committed:
Issue #2876085 by heddn, maxocub, phenaproxima, Jo Fitzgerald, vasi, quietone, yoroy, masipila, larowlan, neclimdul, krystalcode, catch: Before upgrading, audit for potential ID conflicts

larowlan committed:
Issue #2760167 by kim.pepper, markcarver, jibran, dawehner, znerol, Wim Leers, larowlan, xjm, tim.plunkett: Add \Drupal\Core\Messenger\Messenger

Commit:
Issue #2928846 by alexpott, Berdir: [PHP 7.2] count() parameter must be an array or an object that implements Countable

Commit:
Issue #2914938 by timmillwood, RajabNatshah, xjm, Manuel Garcia, amateescu, Wim Leers: Preview of content - Notice: Undefined offset: 0 in _quickedit_entity_is_latest_revision() (line 196 of core/modules/quickedit/quickedit.module)

Commit:
Issue #1489692 by Liam Morland, pfrenssen, YesCT, geekinpink, sudishth, josmera01, David_Rothstein: Incorrect handling of file upload limit exceeded - file widget disappears

Commit:
Issue #2913864 by Jo Fitzgerald, chiranjeeb2410, matslats, phenaproxima: badly constructred link in drupal_set_message

Commit:
Issue #2930072 by vaplas, Lendude: Module: Convert system functional tests to phpunit

xjm committed:
Issue #2926914 by tim.plunkett, xjm, larowlan, tedbow, EclipseGc: Rewrite \Drupal\layout_builder\Section to represent the entire section, not just the block info

effulgentsia committed:
Issue #2929496 by plach, hchonov, timmillwood, catch, Wim Leers, amateescu, Calystod, andypost: Add dedicated interfaces to group methods dealing with revision translation and clean up the related documentation

Commit:
Issue #2551259 by Mile23, gaurav.kapoor, Jo Fitzgerald, googletorp, Jeremy, xjm, webchick: Deprecate dead code locale_translation_manual_status()

effulgentsia committed:
Issue #2825487 by damiankloip, Wim Leers, garphy, cburschka, tedbow, dpovshed, tstoeckler, Munavijayalakshmi, Berdir, dawehner, e0ipso: Fix normalization of File entities: file entities should expose the file URL as a computed property on the 'uri' base field

Commit:
Issue #2862671 by masipila, Jo Fitzgerald, kleog, phenaproxima, quietone: Add documentation to SqlBase source plugin

Commit:
Issue #2921033 by Jo Fitzgerald, masipila, phenaproxima, xjm, Wim Leers: Improve API documentation of DrupalSqlBase source plugin

Commit:
Issue #2931339 by manuel.adan: Unused variable $admin_permission in EntityAccessControlHandler::checkAccess

Commit:
Issue #2932044 by alexpott: Remove \PHPUnit_Util_XML::cssSelect() from \Drupal\tour\Tests\TourTestBase

Commit:
Issue #2884675 by rodrigoaguilera, joelpittet, TR: Remove twig uses of Twig_Node::getLine to Twig_Node::getTemplateLine

Commit:
Issue #2931368 by opdavies: Add missing param documentation for hook_migrate_prepare_row, document hook_migrate_MIGRATION_ID_prepare_row

Commit:
Issue #2927806 by alexpott, mondrake, jibran, Mile23: Use PHPUnit 6 for testing when PHP version >= 7.2

Commit:
Issue #2923015 by cburschka, amateescu, tstoeckler, alexpott, hchonov, mondrake, pfrenssen: [PHP 7.2] Incompatible method declarations

Commit:
Issue #2930788 by marcoscano, phenaproxima, Berdir, seanB: Do not show name by default in media displays

webchick committed:
Issue #2899708 by gaurav.kapoor, tedbow, droplet, Wim Leers: `quote` should be `blockquote` in off-canvas.base.css

larowlan committed:
Issue #2845361 by claudiu.cristea, Munavijayalakshmi: Don't compute children and parents of a new term on TermForm

larowlan committed:
Issue #2346893 by lauriii, idebr, slashrsm, RavindraSingh, Rade, Fabianx, alexpott, swentel, gauravjeet, darrenwh, deepak_zyxware, joelpittet, Wim Leers, Yogesh Pawar, Vj, ivan.chavarro, josephdpurcell, josmera01, rloos289, kattekrab, Tanvish Jha, csakiistvan, xjm, larowlan, akalata: Duplicate AJAX wrapper around a file field

larowlan committed:
Issue #2932551 by jeqq: Error when calling ModerationStateFieldItemList::updateModeratedEntity() if the entity doesn't have workflow

larowlan committed:
Issue #2932154 by jhedstrom: ModerationInformation::getLatestRevisionId returns access-specific results

larowlan committed:
Issue #2931598 by kim.pepper, markcarver, almaudoh, Sam152: Messenger methods drop repeat flag

larowlan committed:
Issue #2928450 by tim.plunkett: Remove dead code in the Layout Builder following Section refactoring

larowlan committed:
Issue #2931264 by markcarver, claudiu.cristea: Remove static \Drupal::$legacyMessenger property

larowlan committed:
Issue #2931709 by marcoscano, claudiu.cristea: Wrong constant name in \Drupal\image\Plugin\Field\FieldType\ImageItem::generateSampleValue()

larowlan committed:
Issue #2931294 by claudiu.cristea, Wim Leers: Timestamp field type misses schema for value

larowlan committed:
Issue #2779921 by kiamlaluno, alexpott: hook_field_widget_form_alter() still reference a hook that is not used anymore

larowlan committed:
Issue #2840257 by kiamlaluno: The documentation makes reference to a function that doesn't exist

larowlan committed:
Issue #2323459 by harsha012, jhodgdon, joachim: Change wording of annotation keys to properties

Commit:
Issue #2862894 by fgm, Wim Leers, bkosborne, borisson_, catch: Docs for Internal Page Cache incorrectly state that it respects the maximum age performance setting

larowlan committed:
Issue #2626924 by Wim Leers, tedbow, dawehner, frob, martin107, damiankloip, dagmar, almaudoh, Berdir, larowlan, amateescu: Include processed text in normalizations: "text" field type's "processed" computed property should be non-internal and carry cacheability metadata

larowlan committed:
Issue #2933125 by Tessa Bakker: Case mismatch in ExportForm.php

Commit:
Issue #2837022 by hchonov, xjm, vlad.dancer, plach, matsbla, Gábor Hojtsy: Concurrently editing two translations of a node may result in data loss for non-translatable fields

effulgentsia committed:
Issue #2824851 by Wim Leers, arshadcn, amateescu, effulgentsia, tedbow, timmillwood, cburschka, tstoeckler, Berdir, xjm, catch: EntityResource::patch() makes an incorrect assumption about entity keys, hence results in incorrect behavior

effulgentsia committed:
Issue #2926483 by plach, amateescu, effulgentsia, xjm, Sam152, timmillwood, jibran, larowlan, Gábor Hojtsy: Add API methods for determining whether an entity object is the latest (translation-affecting) revision

Commit:
Issue #2930715 by alexpott, dawehner: Recursive rebuild caused by installing admin_toolbar_tools module

Commit:
Issue #2932865 by kiamlaluno, joachim: Incorrect description for ExtensionDiscovery::scanDirectory()

Commit:
Issue #2862743 by masipila: Add documentation to DestinationBase destination plugin

Commit:
Issue #2894068 by Jo Fitzgerald, davidsickmiller, alexpott, heddn, Yogesh Pawar, quietone, xjm: datetime_type is not set correctly when migrating datetime fields from D7

Commit:
Issue #2866779 by Mile23, dawehner: Add a way to trigger_error() for deprecated hooks

webchick committed:
Issue #2924351 by drpal, tedbow, xjm, dawehner: Fix coding standards issues with existing settings tray JavaScript

webchick committed:
Issue #2919147 by tedbow, tim.plunkett: When edit mode is enabled new page loads will not have Contextual tabbing constrained

larowlan committed:
Issue #2932774 by harsha012, gaurav.kapoor, neelam.chaudhary, dpi: Update docs for getEmail, can return null

larowlan committed:
Issue #2359389 by harsha012, hgoto, jeqq, fago, sureshcj, larowlan: Call to a member function filters() on a non-object in core/modules/text/text.module on line 83

larowlan committed:
Issue #2928699 by marcoscano, phenaproxima, amateescu, seanB, yoroy: Add an alter hook for the pre-configured field UI options and implement it in the Media module

larowlan committed:
Issue #2926309 by vaplas, alexpott, mpdonadio, tacituseu, Mixologic: Random fail due to APCu not being able to allocate memory

webchick committed:
Issue #2925064 by drpal, dawehner, droplet, xjm, webchick, justafish: [1/2] JS codestyle: no-restricted-syntax

Commit:
Issue #2933769 by masipila: Merge handbook documentation to API: substr process plugin

alexpott committed:
Issue #2377747 by mondrake, eiriksm, rpayanm, adci_contributor, mgifford, oakulm, Truptti, chetan2111, joyceg, alexpott, xjm, yoroy, catch: Incorrect node create validation error when an invalid image is attached to a field

xjm committed:
Issue #2915759 by samuel.mortenson, tedbow, xjm, yoroy: :focus is is hard to see for links in the off-canvas dialog

larowlan committed:
Issue #2918500 by tim.plunkett, EclipseGc, tedbow, larowlan, jibran, Wim Leers, phenaproxima, amateescu, borisson_, samuel.mortenson, gaurav.kapoor, KarlShea, hctom, mroycroft, neerajsingh, DamienMcKenna, dsnopek, Xano, TravisCarden, Tim Bozeman: Create a block which can render entity fields

larowlan committed:
Issue #2933424 by David_Rothstein, harsha012, jhodgdon, vaplas: English-specific links to php.net shouldn't be used in non-translatable strings

larowlan committed:
Issue #2862741 by masipila: Add documentation to ComponentEntityDisplayBase destination plugin

larowlan committed:
Issue #2859381 by Manuel Garcia, Sam152, mstef, jhedstrom: Broken/missing handler for Moderation state field

larowlan committed:
Issue #2934520 by tstoeckler: Avoid information disclosure by timing attack in EntityResource::patch()

alexpott committed:
Issue #2866812 by idebr: Update stylelint rule function-name-case to be consistent with Drupal's CSS standards

alexpott committed:
Issue #2866799 by cwells: Update stylelint rule at-rule-empty-line-before to be consistent with Drupal's CSS standards

alexpott committed:
Issue #2866809 by BrightBold: Update stylelint rule declaration-colon-space-after to be consistent with Drupal's CSS standards

alexpott committed:
Issue #2866800 by BrightBold: Update stylelint rule block-no-empty to be consistent with Drupal's CSS standards

alexpott committed:
Issue #2866808 by BrightBold, pk188, idebr: Update stylelint rule declaration-block-trailing-semicolon to be consistent with Drupal's CSS standards

xjm committed:
Issue #2863354 by benqwerty, DuaelFr, andrewmacpherson, xjm, mgifford, lauriii, ckrina: Add border to dialog [x] close button for hover and focus states

larowlan committed:
Issue #2933991 by alexpott, neclimdul: Deprecation tests fail when all PHPUnit tests are run via PHPUnit

larowlan committed:
Issue #2934152 by Wim Leers, neclimdul: ContentTypeHeaderMatcher should not run for GET, HEAD, OPTIONS or TRACE requests

larowlan committed:
Issue #2931047 by Dropa, D34dMan, borisson_, catch: hook_post_update_NAME documentation is misleading

larowlan committed:
Issue #2861840 by tim.plunkett, lauriii: Preprocess functions are not merged when a module registers a theme hook for a theme-provided template

larowlan committed:
Issue #2934233 by neclimdul: Fix reference to DrupalKernelTest in system module

larowlan committed:
Issue #2929835 by amateescu, dawehner: [regression] Modules can no longer alter the the table queue of a \Drupal\views\Plugin\views\query\Sql query object

larowlan committed:
Issue #2765849 by kiamlaluno: The description for template_preprocess() has not been update for Drupal 8

larowlan committed:
Issue #2933980 by richgerdes, Maheshwaran.j: JQuery UI CSS dependencies are not met in core.libraries.yml

larowlan committed:
Issue #2931883 by moshe weitzman, David_Rothstein, Wim Leers: Unneeded always_populate_raw_post_data requirements check while on CLI

larowlan committed:
Issue #2816861 by dpi, jibran, bucefal91, harsha012, tim.plunkett, catch: Action configuration form does not support #ajax

larowlan committed:
Issue #2856950 by dmsmidt, marcvangend, tim.plunkett, andrewmacpherson, Lendude, tameeshb, Berdir, xjm, tedbow, Gábor Hojtsy, Sutharsan: Add a possibility to disable inline form errors for a complete form

larowlan committed:
Issue #2934517 by Berdir: Setting a revision ID on a new entity sets the newRevision flag to false

alexpott committed:
Issue #2866805 by Mukeysh, joelpittet, harsha012: Update stylelint rule declaration-block-no-redundant-longhand-properties to be consistent with Drupal's CSS standards

xjm committed:
Issue #2932369 by marcoscano, balsama, Berdir, Grimreaper, alexpott: Media Types missing access control handler result in empty column in media overview page

xjm committed:
Issue #2934840 by robpowell, marcoscano: MediaAccessTest wrongly extends another test class, instead of base class

xjm committed:
Issue #2934850 by balsama, Berdir: Media Images should be rendered at a reasonable size by default

xjm committed:
Issue #2924631 by marcoscano, starshaped, chr.fritsch, xjm, phenaproxima, Gábor Hojtsy, seanB, yoroy, evankay: Media sources for local video and audio support

xjm committed:
Issue #2928798 by balsama, axel.rutz, xjm, phenaproxima, chr.fritsch, jeetendrakumar: Add extractors: mimetype & filesize

xjm committed:
Issue #2934962 by starshaped, xjm: Ship local audio and video media types in Standard

larowlan committed:
Issue #2934096 by kiamlaluno, joelpittet: Simplify the ListDataDefinition::getClass() code

xjm committed:
Issue #2932226 by robpowell, marcoscano, phenaproxima, Darvanen, larowlan: Media Type entities don't validate machine name properly

larowlan committed:
Issue #2671964 by EclipseGc, tim.plunkett, Jo Fitzgerald, larowlan, fago, dawehner, Berdir, phenaproxima: ContextHandler cannot validate constraints

larowlan committed:
Issue #2934966 by benjifisher, starshaped: Make it easier to get source field values from media items

larowlan committed:
Issue #2575135 by quietone, ckaotik, olegel, mikeryan, larowlan: Dummy map/message tables being created

larowlan committed:
Issue #2917594 by kevin.dutra, robin.ingelbrecht, jwkovell, Lendude: Add support for HTML5 placeholder in views exposed filters

xjm committed:
Issue #2934649 by chr.fritsch, marcoscano: Fix wrong header "Source" on media overview page

xjm committed:
Issue #2933964 by kiamlaluno, oakulm: Remove the left-over @todo comments in the BaseFieldDefinition class

xjm committed:
Issue #2775131 by chr.fritsch, marcoscano, katzilla, seanB, xjm, gippy, phenaproxima, alexpott, Berdir, lauriii, webflo, dawehner: Media entities should support contextual links

xjm committed:
Revert "Issue #2775131 by chr.fritsch, marcoscano, katzilla, seanB, xjm, gippy, phenaproxima, alexpott, Berdir, lauriii, webflo, dawehner: Media entities should support contextual links"

This reverts commit 1da5a5ccb31a844127998907dddd76667a2bc16c.

xjm committed:
Issue #2775131 by chr.fritsch, marcoscano, katzilla, seanB, xjm, gippy, phenaproxima, alexpott, Berdir, lauriii, webflo, dawehner: Media entities should support contextual links

alexpott committed:
Issue #2866823 by harsha012, Mukeysh, joelpittet: Update stylelint rule selector-type-no-unknown to be consistent with Drupal's CSS standards

xjm committed:
Issue #2934162 by marcoscano, phenaproxima, chr.fritsch, robpowell, Berdir, xjm, effulgentsia: Provide better defaults to media form displays

xjm committed:
Issue #2934499 by samuel.mortenson: Adjust foccus styling for the Settings Tray [x] close button to match the dialog system

alexpott committed:
Issue #2916740 by chr.fritsch, xjm, Manuel Garcia, tstoeckler, alexpott, seanB, Sam152, amateescu, tim.plunkett, dawehner, larowlan, Berdir: Add generic entity actions

Commit:
Issue #2904084 by Ruslan P, Venkatesh Rajan.J: Confusing title for "Boolean" field in "Manage form display"

xjm committed:
Issue #2935318 by marcoscano: Improve default display of Audio and Video media types shipped with Standard profile

alexpott committed:
Issue #2866810 by BrightBold, harsha012, joelpittet: Update stylelint rule function-comma-space-after to be consistent with Drupal's CSS standards

larowlan committed:
Issue #2934424 by chr.fritsch, benjifisher, tstoeckler, phenaproxima, xjm, marcoscano, larowlan: Media has no collection route

larowlan committed:
Issue #2935151 by alexpott: Long array syntax checking in ./core files is broken

larowlan committed:
Issue #2935524 by neclimdul: PHPCS failure on DrupalStandardsListenerTrait

alexpott committed:
Issue #2866816 by harsha012, BrightBold, cwells, joelpittet, idebr: Update stylelint rule length-zero-no-unit to be consistent with Drupal's CSS standards

Commit:
Issue #2934336 by alexpott: Update symfony/phpunit-bridge to the latest released version

webchick committed:
Issue #2853460 by Wim Leers, dawehner, gabesullice, neclimdul, larowlan: Simplify RequestHandler: make it no longer ContainerAware, split up ::handle()

webchick committed:
Issue #2897272 by tedbow, Wim Leers, Jo Fitzgerald, Adita, xjm: Fix module description, hook_help(), and document module scope in *.api.php file

larowlan committed:
Issue #2923004 by tedbow, bircher, samuel.mortenson, alexpott, Wim Leers, axel.rutz, Berdir, borisson_, xjm, tim.plunkett: Add method to check if any overrides are applied to \Drupal\Core\Config\Config

larowlan committed:
Issue #2862248 by m4olivei, Wim Leers, cburschka, larowlan: UnitTestCase::getConfigFactoryStub() doesn't accept dotted keys

larowlan committed:
Issue #2927349 by tim.plunkett: Decouple the Layout Builder UI from entities

larowlan committed:
Issue #2848085 by snehi, rkoller, Sutharsan, Gábor Hojtsy: "Translation update status" in locale.install is not translated

goba committed:
Issue #2933774 by masipila, quietone: Merge handbook documentation to API: migration_lookup process plugin

goba committed:
Issue #2916199 by quietone, heddn: MigrateUpgradeTestBase shouldn't enable modules

Commit:
Issue #2912803 by Jo Fitzgerald, heddn: MigrationPluginManagerInterface is missing createInstancesByTag

webchick committed:
Issue #2926932 by matthiasm11, Wim Leers, vijaycs85: Update CKEditor library to 4.8.0

goba committed:
Issue #2935822 by sonu.raj.chauhan, quietone: Correct test group for two migrate tests

goba committed:
Issue #2930832 by phenaproxima, heddn, larowlan, quietone: Add isAuditable() method to MigrationInterface

goba committed:
Issue #2862662 by masipila, Jo Fitzgerald, heddn, quietone, ohthehugemanatee: Add documentation to EntityContentBase destination plugin

larowlan committed:
Issue #2908282 by heddn, phenaproxima, Jo Fitzgerald, quietone, xjm, larowlan, mikeryan, dipakmdhrm: Throw exception for source plugins without a source_module property

larowlan committed:
Issue #2883892 by Jo Fitzgerald, voleger, ozin, mpdonadio, Taran2L, heddn: Add from/to timezone settings for the "FormatDate process plugin"

larowlan committed:
Issue #2920172 by sushantpaste, dev.patrick, Ashreen Kaur, yoroy, xjm: User Guide link on front page in new Install should  open in new tab

alexpott committed:
Issue #2866819 by harsha012, jpassetti, joelpittet: Update stylelint rule number-no-trailing-zeros to be consistent with Drupal's CSS standards

Commit:
Issue #2935617 by alexpott, chr.fritsch: Move User module's temp stores to core

goba committed:
Issue #2934885 by marcoscano, benjifisher, seanB, starshaped, xjm, chr.fritsch, phenaproxima, Gábor Hojtsy: Document the difference between File, Image, and Media fields

goba committed:
Issue #2918837 by heddn, neclimdul, phenaproxima, quietone, Gábor Hojtsy: SqlBase throws fatal error when extended directly

webchick committed:
Issue #2921626 by tim.plunkett, EclipseGc, larowlan, dead_arm: Add proper context-awareness to Layout Builder

effulgentsia committed:
Issue #2924724 by plach, hchonov, timmillwood, Gábor Hojtsy, amateescu, gabesullice, catch: Add an API to create a new revision correctly handling multilingual pending revisions

xjm committed:
Drupal 8.6.x-dev

xjm committed:
Revert "Issue #2921626 by tim.plunkett, EclipseGc, larowlan, dead_arm: Add proper context-awareness to Layout Builder"

This reverts commit 749f41ff441d796fd828401c50705ebcc256c074.

webchick committed:
Issue #2921626 by tim.plunkett, EclipseGc, larowlan, dead_arm, xjm: Add proper context-awareness to Layout Builder

xjm committed:
Issue #2931165 by heddn, phenaproxima, maxocub, quietone, masipila: Mark Migrate (API) as stable

webchick committed:
Issue #2934178 by samuel.mortenson, DyanneNova, xjm, yoroy, andrewmacpherson: Contrast issues with off-canvas dialog styling

(cherry picked from commit e428304f344d48178db821f480ca01418bab921c)

effulgentsia committed:
Issue #2891215 by plach, timmillwood, hchonov, amateescu, catch, gabesullice: Add a way to track whether a revision was default when originally created

larowlan committed:
Issue #2933770 by masipila, quietone, phenaproxima: Merge handbook documentation to API: static_map process plugin

goba committed:
Issue #2933776 by masipila, quietone: Merge handbook documentation to API: sub_process process plugin

(cherry picked from commit ed35f803c25943d91def78be4936a6657fa6bd84)

goba committed:
Issue #2933773 by masipila, quietone: Merge handbook documentation to API: skip_on_empty process plugin

(cherry picked from commit 36ea4d333d34a4154b4da6d8110325f2d3a53e46)

goba committed:
Issue #2862748 by masipila, Jo Fitzgerald, quietone, phenaproxima: Add documentation to EntityRevision destination plugin

(cherry picked from commit 253b1020fce1f1a34290287fc53bbd1969342953)

goba committed:
Issue #2862744 by masipila, heddn: Add documentation to EntityBaseFieldOverride destination plugin

(cherry picked from commit a0f31bc76352b718237048ef83668113a35883a1)

larowlan committed:
Issue #2862041 by Sam152, amateescu, timmillwood, jibran, hamrant, dawehner, plach, Manuel Garcia, Jo Fitzgerald, mgalalm, xjm, larowlan, webchick, Wim Leers, gambry, kkus, tim.plunkett, tedbow, CatherineOmega: Provide useful Views filters for Content Moderation State fields

alexpott committed:
Issue #2885469 by Wim Leers, amateescu, timmillwood, alexpott, tstoeckler, xjm, remram, Berdir, dawehner, larowlan, samuel.mortenson: Regression: manually setting the ID field for newly-created content entities is not possible anymore (public follow-up to SA-2017-002)

alexpott committed:
Issue #2936435 by Berdir, alexpott: Better BC for deprecated container parameter user.tempstore.expire

larowlan committed:
Issue #2914974 by quietone, rakesh.gectcr, maxocub, heddn, Jo Fitzgerald, Yogesh Pawar, masipila, phenaproxima, jhodgdon, Gábor Hojtsy, larowlan, neclimdul: Migrate UI - handle sources that do not need an upgrade

larowlan committed:
Issue #2935725 by markconroy: Add vw and vh to the CSS stylelint

larowlan committed:
Issue #2858482 by Wim Leers, dawehner, benjy, larowlan, tedbow, borisson_: Simplify REST routing: disallow requesting POST/PATCH in any format, make consistent

larowlan committed:
Issue #2935755 by alexpott, Wim Leers: Add a trait to allow dynamic setting of expected deprecations

larowlan committed:
Issue #2862422 by chr.fritsch, robpowell, phenaproxima, webflo, bdimaggio, seanB, naveenvalecha, sunset_bill, vijaycs85, yoroy, starshaped, xjm, Berdir, hazong: Add per-media type creation permissions for media

alexpott committed:
Issue #2936457 by Berdir, larowlan: views_post_update_filter_placeholder_text() can result in errors in filter plugins due to incomplete definition

webchick committed:
Issue #2902187 by amateescu, timmillwood, Sam152, webchick, Manuel Garcia, xjm, plach, DuneBL, larowlan, Bojhan, jibran, Berdir, jojototh: Provide a way for users to moderate content

webchick committed:
Issue #2935779 by tim.plunkett, EclipseGc, webchick, xjm, dead_arm: Come up with an affordance for what parts of the page are/are not affected by Layout Builder

webchick committed:
Issue #2936793 by tim.plunkett, tstoeckler, EclipseGc: EntityReferenceItem::generateSampleValue() should create a sample entity if a referenceable entity is not found

larowlan committed:
Issue #2543726 by damiankloip, Wim Leers, claudiu.cristea, voleger, dagmar, dawehner, amateescu, tedbow, FredCorreia, mgalalm, caseylau, jibran, dbjpanda, pcambra, Grimreaper, idebr, dmouse, larowlan, Berdir, chx, andypost, yched, fgm: Make $term->parent behave like any other entity reference field, to fix REST and Migrate support and de-customize its Views integration

webchick committed:
Issue #2936793 follow-up by tim.plunkett: Fix PHP5's hacking and wheezing.

(cherry picked from commit 02af9bc3f6799eea7ac23086e962e603f6bc1d4b)

Commit:
Issue #2908952 by sahil432, martin107: drupal_get_complete_schema() and drupal_get_schema() are not completely removed

Commit:
Issue #2934997 by benjifisher, Lendude, xjm: Intermittent failure in MediaUiJavascriptTest

Commit:
Issue #2936941 by alexpott: Diff can infinitely loop since #2885309

larowlan committed:
Issue #2936802 by alexpott, Wim Leers: expectDeprecation() doesn't work in isolated tests

larowlan committed:
Issue #2927746 by alexpott, catch, slasher13, mpdonadio, martin107, RoSk0, jibran, larowlan, pounard, Mile23, Jo Fitzgerald, deviantintegral, pritish.kumar: Update Symfony components to 3.4.*

webchick committed:
Issue #2809635 by markconroy, navneet0693, alexpott, smaz, larowlan, Eli-T, webchick, yoroy, ikit-claw, budda, kjay, timmillwood, tkoleary, Gábor Hojtsy, andrewmacpherson, lauriii, JayKandari, jibran, ok_lyndsey, big_man, ckrina, nathancz, Paul_Gregory, vijaycs85, SharJay, tomphippen, tim.plunkett, Tarun Lewis, cehfisher, kreynen, Petr Illek, mariohernandez, thamas, thatdamnqa, John Cook, philipnorton42, h2cm, kattekrab, martin_q, waako: Create experimental installation profile

larowlan committed:
Issue #2937894 by navneet0693: Remove unneeded schema entry from demo_umami.schema.yml

xjm committed:
Issue #2937889 by larowlan, Sam152: Regression: LinkWidget no longer supports base fields with optional title

(cherry picked from commit 7871cc00ee694a6f639ce58b07acf5499795aa9b)

webchick committed:
Issue #2919373 by tedbow, Wim Leers, Adita, webchick, Berdir, drpal, xjm, samuel.mortenson, bircher: Prevent Settings Tray functionality for blocks that have configuration overrides

goba committed:
Issue #2878556 by plach, matsbla, vijaycs85, Gábor Hojtsy, catch, gabesullice, effulgentsia, hchonov, hass, amateescu, xjm: Ensure that changes to untranslatable fields affect only one translation in pending revisions

Commit:
Issue #2930542 by sahil432, sdewitt, andrewmacpherson: Disabled form labels in Seven theme's node meta-data  do not meet WCAG AA 1.4.3 Contrast (minimum)

Commit:
Issue #2938360 by markconroy: Change hook_preprocess_block to use $variables instead of $vars

Commit:
Issue #2937636 by piyuesh23, markconroy: Umami Theme - follow-up - Use hook_form_alter() for search block placeholder

Commit:
Issue #2935254 by kim.pepper: Remove all usages of drupal_get_message and drupal_set_message in core includes

Commit:
Issue #2936525 by priya.chat, benjifisher: Confusing documentation for setConfigOverrideLanguage()

Commit:
Issue #2935575 by gaurav.kapoor, kiamlaluno, amateescu: Fix or remove the @see tag

Commit:
Issue #2936641 by alokvermaei: 'Path alias' in PathFilterForm.php is not translated

Commit:
Issue #2914834 by snehi, tresti88, joachim: missing @return for drupal_get_user_timezone() and return types for other methods in bootstrap.inc

Commit:
Issue #2921706 by Daniel Korte, bkosborne: Update example.settings.local.php to include ['cache']['bins']['page']

alexpott committed:
Issue #2938369 by Mohit Malik, yo30, mondrake, alexpott: The Symfony\Component\ClassLoader\WinCacheClassLoader class is deprecated

Commit:
Issue #2872355 by sarguna raj M, chiranjeeb2410, joachim: expand and clarify docs for DefaultPluginManager::alterInfo()

Commit:
Issue #2925997 by tikaszvince, ndolley: position-min.js issue since 8.4 upgrade

larowlan committed:
Issue #2938187 by markconroy, andrewmacpherson: Homepage/Recipes page banner image is duplicating on small screens

larowlan committed:
Issue #2935137 by el7cosmos, Mixologic: License "GPL-2.0+" is a deprecated SPDX license identifier, use "GPL-2.0+-only" or "GPL-2.0+-or-later" instead

Commit:
Issue #2920536 by Wim Leers, Grayside: Force all serializer encoders + normalizers services to be private

Commit:
Issue #2924165 by TravisCarden: Doxygen for callable arguments missing callable keyword

Commit:
Issue #2915782 by voleger, Wim Leers: Modernize ForumTest: use entity queries instead of DB queries

Commit:
Issue #2938799 by claudiu.cristea: Provide the timestamp scalar data type

dawehner committed:
Apply latest version of 2926633

dawehner committed:
adapt the tests

dawehner committed:
Merge branch '8.6.x' into nightwatch

dawehner committed:
Allow to configure the DB URL in the settings.json file

dawehner committed:
Ensure env variables win

dawehner committed:
Allow to override everything using environment variables

ryanaslett committed:
Update nightwatch.conf.js

the ./core directory was redundant

ryanaslett committed:
Update env.js

small typo

dawehner committed:
Merge pull request #5 from jsdrupal/env-variables

Allow to override everything using environment variables

ryanaslett committed:
Update YamlTest.php

This test fails if we've run yarn install and created a node_modules dir because it finds some modules' travis.yml configs etc.

ryanaslett committed:
Revert previous change.

Reverts previous change.

ryanaslett committed:
Un-dumbs da dumb

Accidentally removed an extra line.

dawehner committed:
do some additional checking

ryanaslett committed:
Manual revert

I am so stepping on my own feet today. Best not to manually revert.
